scio-bigquery/src/test/scala/com/spotify/scio/bigquery/types/BigDecimalConversionTest.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.bigquery.types
+
+import com.spotify.scio.bigquery.BigQueryUtil
+import org.scalatest.{FlatSpec, Matchers}
+
+class BigDecimalConversionTest extends FlatSpec with Matchers {
+  case class BigBoy(x: BigDecimal)
+
+  "big decimal values" should "roundtrip if they fit NUMERIC scale" in {
```

Comment:
@lazyval we can do property based testing for this. Maybe we can add the test under `ConverterProviderSpec`.

Reply:
Yeah, I saw it. But it's key idea is that conversions are lossless, whereas bigdecimal is lossy. I can't take an arbitrary BD and expect it to roundtrip just fine: some will not. Limiting the scope of `Arbitrary[BigDecimal]` sounds like a cheating.

Reply:
yeah, I think it's ok to just add a custom `Gen` that would generate numbers up to the same scale/precision as BQ `Numeric` and still keep an extra test that verifies the round up.